### PR TITLE
PerfdataValue: add UoMs

### LINKED ISSUE #7225: Allow more UoMs for perfdata
## Is your feature request related to a problem? Please describe.
Currently we support the UoMs for perfdata as described in the [Monitoring Plugins Guidelines](https://www.monitoring-plugins.org/doc/guidelines.html#AEN201). The guidelines are old and lack a lot of UoM's that should be supported by Icinga. The reason why we pay attention to the UoM is because some plugins change their perfdata output depending on the collected metric. Therefore we convert some values to a basic unit (eg. we convert `kb` to `bytes`)

## Describe the solution you'd like
My suggestion is to extend our support for UoM's and support the following units:
(Some of them are already supported)

|  Category       | UoM                          | Description |
|--------------|---------------------|-------------|
| Data               |                                    |                      |
|                        | b                                 | Bit                |
|                        | B                                 | Byte             |
|                        | KB                              | Kilobyte        |
|                        | MB                             | Megabyte    |
|                        | GB                              | Gigabyte     |
|                        | TB                              | Terrabyte    |
|                        | PB                              | Petabyte      |
|                        | KiB                             | Kibibyte       |
|                        | MiB                            | Mebibyte     |
|                        | GiB                            | Gibibyte       |
|                        | TiB                             | Tebibyte      |
|                        | PiB                             | Pebibyte     |
| Data Rate       |                                    |                      |
|                        | packets                     |  |
|                        | bits                            |  |
|                        | bytes                         | |
|                        | kilobytes                   | |
|                        | kilobits                      | |
|                        | megabytes               | |
|                        | megabits                  | |
|                        | gigabytes                 | |
|                        | gigabits                    | |
|                        | kbit                           | |
|                        | kbyte                        | |
|                        | mbit                          | |
|                        | mbyte                       | |
|                        | gbit                           | |
|                        | gbyte                        | |
| Time               |                                    |                      |
|                        | ns                              | Nanoseconds |
|                        | µs                              | Microseconds |
|                        | ms                             | Milliseconds |
|                        | s                                | Seconds |
|                        | m                               | Minutes |
|                        | h                                | Hours |
|                        | d                                | Days |
| Energy           |                                   |                        |
|                        | mW                           | Milliwatt         |
|                        | W                              | Watt               |
|                        | kW                            | Kilowatt |
|                        | Wh                            | Watt-hour |
|                        | Wm                           | Watt-minute |
|                        | mA                            | Milliampere |
|                        | A                               | Ampere |
|                        | kA                             | Kiloampere |
|                        | mV                            | Millivolt |
|                        | V                               | Volt |
|                        | kV                             | kiloVolt |
|                        | Lm                            | Lumens |
|                        | _O_                            | Ohm |
|                        | dBm                          | Decibel-Milliwatt |
| Mass              |                                     |                      |
|                        | mg                            | Milligram |
|                        | g                               | Gram |
|                        | kg                             | Kilogram |
| Length           |                                    |                      |
|                        | mm                           | Millimeter |
|                        | cm                            | Centimeter |
|                        | m                              | Meter |
|                        | km                            | Kilometer |
|                        | mi                             | Miles |
|                        | ft                              | Feet |
| Temperature |                                    |                      |
|                        | C / °C                       | Celsius |
|                        | F / °F                        | Farenheit |
|                        | K                               | Kelvin |
| Volume          |                                    |                      |
|                        | mL                             | Milliliters |
|                        | L                                | Liters |
| Throughput  |                                     |                      |
|                        | ops                           | ops/sec |
|                        | rps                           | requests/sec |
|                        | wps                          | writes/sec |
|                        | iops                          | I/O ops/sec |
|                        | rpm                          | reads/min |
|                        | wpm                         | writes/min |

Independently from the UoM,  we should always allow a time unit such as `*/s`, `*/sec`, `*/m`, `*/min`, `*/h`, `*/hour`, `*/d` and `*/day`

Instead of dropping values with unknown UoM's, we should rather just pass the values and give the user at least the chance to view the values.

## Things to consider
* We can not change the current conversion of values, this may break graphs in existing setups
* We need to find and verify a suitable conversion for each UoM
* I'm aware that there are tons of other UoM's, this should be a decent extension for the current state. We may consider more UoM's later.

## Additional context
Referred issues:
* https://github.com/Icinga/icinga2/issues/6849
* https://github.com/Icinga/icinga2/issues/6867

/cc @dnsmichi @lippserd

### PR DESCRIPTION
**Pull request by Al2Klimov**
```
* {,{K,M,G,T,P,E,Z,Y}{,i}}B
  => bytes
* {,{k,m,g,t,p,e,z,y}{,i}}b
  => bits
* packets
* {n,u,m,}s m h d
  => seconds
* {n,u,m,,k,M,G,T,P,E,Z,Y}{{A,O,V,W},{A,W}{s,m,h}}
  => amperes, ohms, volts, watts, ampere-seconds, watt-hours
* lm dBm
  => lumens decibel-milliwatts
* {n,u,m,,k}g t
  => grams
* C F K
  => degrees-celsius, degrees-farenheit, degrees-kelvin
* {m,,h}l
  => liters
```

fixes #7225
closes #6868

### CODE REVIEW

**Comment by Al2Klimov** [comment]
@N-o-X Please have a look whether that set of units is OK for you.

**Comment by Al2Klimov** [comment]
FYI @lippserd @LordHepipud @Thomas-Gelf

File: lib/base/perfdatavalue.cpp [hunk 1]
```diff
@@ -6,12 +6,210 @@
...
+		// Volume:
+		{ "ml", { 1.0 / 1000, "liters" } },
+		{ "l", { 1, "liters" } },
+		{ "hl", { 100, "liters" } },
+
+		// Misc:
+		{ "packets", { 1, "packets" } },
+		{ "lm", { 1, "lumens" } },
+		{ "dbm", { 1, "decibel-milliwatts" } },
+		{ "f", { 1, "degrees-farenheit" } },
```
Comment (N-o-X) [inline lib/base/perfdatavalue.cpp]: *fahrenheit

File: doc/05-service-monitoring.md [hunk 2]
```diff
@@ -646,22 +646,46 @@ Icinga sets `LC_NUMERIC=C` to enforce this locale on plugin execution.
...
+watt-hours         | nWh, uWh, mWh, Wh, kWh, MWh, GWh, ..., YWh, all of these also for Wm and Ws
+lumens             | lm
+decibel-milliwatts | dBm
+grams              | ng, ug, mg, g, kg, t
+degrees-celsius    | C
+degrees-fahrenheit | F
+degrees-kelvin     | K
+liters             | ml, l, hl
+
+Bad plugins change the UOM for different sizing, e.g. returning the disk usage in MB and later GB
```
Comment (N-o-X) [inline doc/05-service-monitoring.md:681]: - I'd not call a plugin `bad`, that's a bit to judgmental for a documentation in my opinion. Maybe just use `some`?
- UoM not UOM (typo)

**Pull request merged**